cli/azd/pkg/apphost/generate.go

```diff
@@ -369,6 +430,58 @@ func (b *infraGenerator) LoadManifest(m *Manifest) error {
 				// a part of a server and it should not be created as a separate resource.
 				b.addContainerAppService(name, "postgres")
 			}
+		case "parameter.v0":
+			b.addInputParameter(name, comp.Inputs["value"].Type)
+		case "azure.bicep.v0":
+			if comp.Path == nil {
+				if comp.Parent == nil {
+					return fmt.Errorf("bicep resource %s does not have a path or a parent", name)
+				}
+				// module uses parent
+				continue
+			}
+			if comp.Params == nil {
+				comp.Params = make(map[string]any)
+			}
+			useSecretOutputs := false
+			if comp.ConnectionString != nil && strings.Contains(*comp.ConnectionString, "secretOutputs") {
+				useSecretOutputs = true
+			}
+			keyVaultForParam := ""
+			emptyJsonString := "\"\""
+			for p, pVal := range comp.Params {
+				jsonBytes, err := json.Marshal(pVal)
+				finalParamValue := string(jsonBytes)
+				if err != nil {
+					return fmt.Errorf("marshalling param %s. error: %w", p, err)
+				}
+				if p == "keyVaultName" && useSecretOutputs {
+					keyVaultForParam = name + "kv"
```

Comment:
Who is responsible for spinning up this KeyVault?  I would have assumed it was `azd` - but a glance over `resources.bicept` didn't show new code that would allocate it.  Did we already have this or am I looking in the wrong place for it?

Reply:
I don't see it either, but I see it working while using it.

Reply:
Yes - interesting.  I finally found the time to pull this work and run it against the sample in the Aspire repo and I see the keyvault being created in `resources.bicep` so I guess this is all working correctly, I just need to read the code more to better understand what is going on.

One thing I did notice, @vhvb1989, and this is likely an artifact of us re-using our keyvault logic, we end up with something that looks like this:

```bicep
resource cosmoskv 'Microsoft.KeyVault/vaults@2023-07-01' = {
  name: replace('cosmoskv-${resourceToken}', '-', '')
  location: location
  properties: {
    sku: {
      name: 'standard'
      family: 'A'
    }
    tenantId: subscription().tenantId
    enableRbacAuthorization: true
  }
  tags: union(tags, {'aspire-resource-name': 'cosmoskv'})
}
```

I don't think we want to have the `aspire-resource-name` tag here - there is no `cosmoskv` resource in the aspire resource model, so I think noting it in this way is wrong (similar to how we don't add these tags to the other resources we create on behalf of the user like the ACR and the ACA Managed Environment).

Reply:
Should we add a tag like `generated-kv-for-cosmos-service` ?

I think the tag is a good idea to anounce why that kv is there.  That was the reason why I added `kv` after the service that is using it.
I could also rename it to `cosmos-aux-kv-{hash}`

Reply:
If we're going to move away from a keyvault per resource then I'm not sure it's worth refactoring yet.

Reply:
tags removed from these keyvaults